clj/src/cljd/core.cljd

```diff
@@ -7194,7 +7194,49 @@
       ([] (gensym "G__"))
       ([prefix-string] (symbol (str prefix-string (swap! id inc)))))))
 
-(def tagged-literal nil) ; TODO hack for macros but agged-literal will have to be implemented in cljd
+(deftype TaggedLiteral [tag form]
+  Object
+  (toString [coll]
+    (pr-str coll))
+
+  IEquiv
+  (-equiv [this other]
+    (and (dart/is? other TaggedLiteral)
+         (= tag (.-tag other))
```

Comment:
Should `other` here be typed hinted as a `TaggedLiteral`?

Reply:
yes it should

Reply:
e.g: `(.-tag ^TaggedLiteral other)`